api/block.go

```diff
@@ -103,7 +156,28 @@ func (api *BlockAPI) GetBlockByHash(ctx context.Context, blockHash types.Hash, i
 	return blockVal, nil
 }
 
+var (
+	gtcbnHitMeter = metrics.NewMinorMeter("/flume/gtcbn/hit")
+	gtcbnMissMeter = metrics.NewMinorMeter("/flume/gtcbn/miss")
+)
+
 func (api *BlockAPI) GetBlockTransactionCountByNumber(ctx context.Context, blockNumber vm.BlockNumber) (hexutil.Uint64, error) {
+
+	if len(api.cfg.HeavyServer) > 0 && !blockDataPresent(blockNumber, api.cfg, api.db) {
+		log.Debug("eth_getBlockTransactionCountByNumber sent to flume heavy")
+		missMeter.Mark(1)
+		gtcbnMissMeter.Mark(1)
+		count, err := heavy.CallHeavy[hexutil.Uint64](ctx, api.cfg.HeavyServer, "eth_getBlockTransactionCountByNumber", hexutil.Uint64(blockNumber))
```

Comment:
I think you want to leave blockNumber as a vm.BlockNumber here, which could leave in values like "latest", which is not going to convert to a hexutil.Uint64 cleanly.